src/main/java/com/onlinebookstore/onlinebookstore/repository/impl/BookRepositoryImpl.java

```diff
@@ -56,6 +59,8 @@ public Optional<Book> findById(Long id) {
         try (Session session = sessionFactory.openSession()) {
             Book book = session.find(Book.class, id);
             return Optional.ofNullable(book);
+        } catch (Exception e) {
+            throw new EntityNotFoundException("Can't find book by id " + id);

```

Comment:
we can have all sorts of exceptions caught here, and not every one of them would mean that an entity with such an id does not exist. We should throw a simple RuntimeEx or something like `DataProcessingException` like we had before